exaudfclient/base/javacontainer/test/cpp/javavm_test.cc

```diff
@@ -0,0 +1,21 @@
+#include "javacontainer/test/cpp/javavm_test.h"
+#include "javacontainer/javacontainer.h"
+#include "exaudflib/exaudflib.h"
+
+
+JavaVMTest::JavaVMTest(std::string scriptCode) : javaVMInternalStatus() {
+    char* script_code = ::strdup(scriptCode.c_str());
+    SWIGVMContainers::SWIGVM_params->script_code = script_code;
```

Comment:
Do I understand the purpose of the query correct, when I assume that you wanted to encapsulate the setting of the script code?

Reply:
This class (JavaVMTest) is just a wrapper; it

1. Sets the swigvm_parameters (only script_code actually needed)
2. Instantiates the JavaVMIMpl
3. and copies the internal members.
Note that JavaVMTest has been declared as friend class of JavaVMImpl in order to get access to it's private members.